lib/matplotlib/tests/test_image.py

```diff
@@ -1468,3 +1468,15 @@ def test__resample_valid_output():
         resample(np.zeros((9, 9), np.uint8), np.zeros((9, 9)))
     with pytest.raises(ValueError, match="must be C-contiguous"):
         resample(np.zeros((9, 9)), np.zeros((9, 9)).T)
+
+
+def test_axesimage_get_shape():
+    # generate dummy image to test get_shape method
+    ax = plt.gca()
+    im = AxesImage(ax)
+    with pytest.raises(RuntimeError):
```

Comment:
```suggestion
    with pytest.raises(RuntimeError, match="You must first set the image array"):
```
to make sure it is the "correct" RuntimeError.